### PR TITLE
fix(portal-backend): introduce ASGARDEO_ISSUER to configure custom ExpectedIssuer for standard OIDC IdPs

### PR DESCRIPTION
**Pull request by ginaxu1**
Resolves https://github.com/LSFLK/lsf-govtech-ndx/issues/5

## Summary
Introduce the `ASGARDEO_ISSUER` env variable to allow setting a custom expected issuer claim (`iss`) for the JWT validation middleware. Previously, the backend validated the issuer claim strictly against `ASGARDEO_TOKEN_URL` (the token endpoint), which broke compatibility with standard OIDC identity providers that emit the base realm/tenant URL as the issuer claim

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## Changes Made

- Modified [portal-backend/main.go](file:///Users/tmp/opendif-core/portal-backend/main.go#L80): Configured `ExpectedIssuer` to look up `ASGARDEO_ISSUER` before falling back to `ASGARDEO_TOKEN_URL`
- Modified [portal-backend/.env.example](file:///Users/tmp/opendif-core/portal-backend/.env.example#L31): Added documentation for the optional `ASGARDEO_ISSUER` variable

## Testing

- [x] I have tested this change locally (verified that the unit tests in the middleware package compile and pass successfully)
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Additional Notes

This fix retains full backward compatibility for existing deployments using Asgardeo, which will continue to default to the token endpoint issuer.

## Deployment Notes

Deployers using standard OIDC identity providers should configure the `ASGARDEO_ISSUER` environment variable in their setup with their identity provider's realm or tenant URL

### CODE REVIEW

File: portal-backend/main.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func main() {
 
 	jwtConfig := v1middleware.JWTAuthConfig{
 		JWKSURL:        utils.GetEnvOrDefault("ASGARDEO_JWKS_URL", asgardeoBaseURL+"/oauth2/jwks"),
-		ExpectedIssuer: utils.GetEnvOrDefault("ASGARDEO_TOKEN_URL", asgardeoBaseURL+"/oauth2/token"),
+		ExpectedIssuer: utils.GetEnvOrDefault("ASGARDEO_ISSUER", utils.GetEnvOrDefault("ASGARDEO_TOKEN_URL", asgardeoBaseURL+"/oauth2/token")),
```
[the remark this answers is not in the document]
Comment (ginaxu1) [reply portal-backend/main.go:90]: addressed https://github.com/OpenDIF/opendif-core/pull/458/changes/010dc1f042269a2b885e327bc1c112833be8ce80
